### ISSUE TITLE
from __future__ import annotations breaks recursive TypeVar

### ISSUE THREAD

**Issue by QuentinSoubeyran** [microsoft/pyright#2093]
**Bug description**
Code that type-checks no longer passes if  `from __future__ import annotations` is used. I believe the future import is only a runtime feature, which doesn't change the semantics of the type annotations, so it shouldn't affect the result of static type-checking.

**To Reproduce**
+ Type-check the code below with pyright (passes)
+ Uncomment the `__future__` import
+ Type-check again, pyright reports `TypeVar "RecDict" is not subscriptable`.

**Expected behavior**
Type-checking succeeds with or without the future import

**Code**
```python
#from __future__ import annotations

from typing import Dict, TypeVar, Union

K = TypeVar("K")
V = TypeVar("V")
RecDict = Dict[K, Union[V, "RecDict[K, V]"]]

def foo(x: RecDict[str, int]) -> RecDict[str, str]:
    return {}
```

**VS Code extension or command-line**
Tested on:
+ VSCode "Pyright" latest extension (v1.1.150)
+  latest command-line (v1.1.156)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This fix is included in pyright 1.1.57, which I just published. It will also be included in the next release of pylance.
